Mathlib/NumberTheory/EulerProduct/Basic.lean

```diff
@@ -0,0 +1,202 @@
+/-
+Copyright (c) 2023 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.Analysis.Normed.Field.InfiniteSum
+import Mathlib.Analysis.SpecificLimits.Normed
+import Mathlib.NumberTheory.ArithmeticFunction
+import Mathlib.NumberTheory.SmoothNumbers
+
+/-!
+# Euler Products
+
+The main result in this file is `EulerProduct.eulerProduct`, which says that
+if `f : ℕ → R` is norm-summable, where `R` is a complete normed commutative ring and `f` is
+multiplicative on coprime arguments with `f 0 = 0`, then
+`∏ p in primesBelow N, ∑' e : ℕ, f (p^e)` tends to `∑' n, f n` as `N` tends to infinity.
+`Nat.ArithmeticFunction.IsMultiplicative.eulerProduct` is a version of
+`EulerProduct.eulerProduct` for multiplicative arithmetic functions in the sense of
+`Nat.ArithmeticFunction.IsMultiplicative`.
+
+There is also a version `EulerProduct.eulerProduct_completely_multiplicative`, which states that
+`∏ p in primesBelow N, (1 - f p)⁻¹` tends to `∑' n, f n` as `N` tends to infinity,
+when `f` is completely multiplicative with values in a complete normed field `F`
+(implemented as `f : ℕ →*₀ F`).
+
+An intermediate step is `EulerProduct.summable_and_hasSum_smoothNumbers_prod_primesBelow_tsum`
+(and its variant `EulerProduct.summable_and_hasSum_smoothNumbers_prod_primesBelow_tsum_geometric`),
+which relates the finite product over primes to the sum of `f n` over `N`-smooth `n`.
+
+## Tags
+
+Euler product
+-/
+
+/-- If `f` is multiplicative and summable, then its values at natural numbers `> 1`
+have norm strictly less than `1`. -/
+lemma Summable.norm_lt_one {F : Type*} [NormedField F] [CompleteSpace F] {f : ℕ →* F}
+    (hsum : Summable f) {p : ℕ} (hp : 1 < p) :
+    ‖f p‖ < 1 := by
+  refine summable_geometric_iff_norm_lt_1.mp ?_
+  simp_rw [← map_pow]
+  exact hsum.comp_injective <| Nat.pow_right_injective hp
+
+
+open scoped Topology
+
+open Nat BigOperators
+
+lemma Nat.map_prime_pow_mul {F : Type*} [CommSemiring F] {f : ℕ → F}
```

Comment:
This should probably move to another file?
Even the lemma above I'm not sure about, but seems more specialist so I'd be happy if there was no other good place for that one

Reply:
OK, moving it into `Mathlib.NumberTheory.SmoothNumbers`.

Regarding the first lemma, `#find_home!` suggests `Mathlib.Analysis.SpecificLimits.Normed` (where `summable_geometric_iff_norm_lt_1` is stated, which is used in the proof). But I think it doesn't fit so well there, since it is about a monoid homomorphism.

Reply:
Yeah I figure anything making use of both additive and multiplicative structures of the naturals belongs in number theory somewhere anyway ;)